File: tournaments/arrayMaximalAdjacentDifference/arrayMaximalAdjacentDifference.js
function arrayMaximalAdjacentDifference(inputArray) {
    let result = 0;

    for (let i = 1; inputArray.length > i; ++i) {
        result = Math.max( result, Math.abs(inputArray[i] - inputArray[i - 1]));
    }

    return result;
}

File: tournaments/arrayMaximalAdjacentDifference/arrayMaximalAdjacentDifference.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub arrayMaximalAdjacentDifference {
    my ($inputArray) = @_;

    use List::Util qw(max);

    my $result = 0;

    for (my $i = 1; @$inputArray > $i; ++$i) {
        $result = max($result, abs($$inputArray[$i] - $$inputArray[$i - 1]));
    }

    return $result;
}
